lsp-ui-sideline.el

```diff
@@ -386,45 +386,51 @@ to the language server."
+    (let ((eol (line-end-position))
+          (bol (line-beginning-position))
+          (line (line-number-at-pos))
+          (doc-id (lsp--text-document-identifier)))
       (save-excursion
         (setq lsp-ui-sideline--occupied-lines nil
               lsp-ui-sideline--tag (lsp-ui-sideline--calculate-tag)
               lsp-ui-sideline--last-width (window-text-width))
-        (goto-char bol)
         (when lsp-ui-sideline-show-flycheck
           (lsp-ui-sideline--flycheck))
         (when (and lsp-ui-sideline-show-code-actions (lsp--capability "codeActionProvider"))
           (lsp--send-request-async (lsp--make-request
                                     "textDocument/codeAction"
-                                    action-params)
+                                    (if (equal lsp-ui-sideline-update-mode 'line)
+                                        (list :textDocument doc-id
+                                              :range (lsp--region-to-range bol eol)
+                                              :context (list :diagnostics (lsp--cur-line-diagnotics)))
+                                      (lsp--text-document-code-action-params)))
                                    #'lsp-ui-sideline--code-actions))
-        (when lsp-ui-sideline-show-hover
-          (while (and (<= (point) eol) (< (point) eob))
-            (let ((symbol (thing-at-point 'symbol t))
-                  (bounds (bounds-of-thing-at-point 'symbol))
-                  (column (lsp--cur-column))
-                  (in-string (nth 3 (syntax-ppss))))
-              ;; Skip string and C++ scope resolution operator ::
-              (when (and symbol (not in-string) (not (looking-at-p "::")))
+        ;; Go through all symbols and request hover information.  Note that the symbols are
+        ;; traversed backwards as `forward-symbol' with a positive argument will jump just past the
+        ;; current symbol.  By going from the end of the line towards the front, point will be placed
+        ;; at the beginning of each symbol.  As the requests are first collected in a list before
+        ;; being processed they are still sent in order from left to right.
+        (when (and lsp-ui-sideline-show-hover (lsp--capability "hoverProvider"))
+          (let ((symbols))
+            (goto-char eol)
+            (while (and (> (point) bol)
+                        (progn (forward-symbol -1)
+                               (>= (point) bol)))
+              (let* ((symbol (thing-at-point 'symbol t))
+                     (bounds (bounds-of-thing-at-point 'symbol))
+                     (parsing-state (syntax-ppss))
+                     (in-string (nth 3 parsing-state))
+                     (outside-comment (eq (nth 4 parsing-state) nil)))
+                ;; Skip strings and comments
+                (when (and symbol (not in-string) outside-comment)
+                  (push (list symbol line bounds (lsp--cur-position)) symbols))))
```

Comment:
`lsp--cur-position` is using `line-number-at-pos` which can be very slow on big files.
Can you use the same logic that was done ? (Computing the current line before looping on every symbol and use it to send the position)